### PR TITLE
feat(WebGPU): add some support for texture mapping

### PR DESCRIPTION
**Pull request by martinken**
Can now do interpolatebefore using textures
and can handle simple textue mapping cases.

Adjusted caches to use WeakMaps

### CODE REVIEW

**Comment by martinken** [comment]
@jourdain PTAL

File: Sources/Rendering/WebGPU/BufferManager/index.js [hunk 1]
```diff
@@ -465,7 +495,26 @@ function vtkWebGPUBufferManager(publicAPI, model) {
     }
 
     buffer.setSourceTime(req.time);
-    model.buffers.push({ request: req, buffer });
+
+    // cache the buffer if we have a dataArray.
+    // We create a new req that only has the 4 fields required for
+    // a comparison to avoid GC cycles
+    if (Object.prototype.hasOwnProperty.call(req, 'dataArray')) {
```
Comment (jourdain) [inline Sources/Rendering/WebGPU/BufferManager/index.js:502]: This seems convoluted...

File: Sources/Rendering/WebGPU/Sampler/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  vtkWebGPUSampler(publicAPI, model);
+}
+
+// ----------------------------------------------------------------------------
+
+export const newInstance = macro.newInstance(extend);
+
+// ----------------------------------------------------------------------------
+
+export default { newInstance, extend, ...STATIC };
```
Comment (jourdain) [inline Sources/Rendering/WebGPU/Sampler/index.js:61]: You don't have anything static, you can remove it...

File: Sources/Rendering/WebGPU/Texture/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  vtkWebGPUTexture(publicAPI, model);
+}
+
+// ----------------------------------------------------------------------------
+
+export const newInstance = macro.newInstance(extend);
+
+// ----------------------------------------------------------------------------
+
+export default { newInstance, extend, ...STATIC };
```
Comment (jourdain) [inline Sources/Rendering/WebGPU/Texture/index.js:145]: no static

File: Sources/Rendering/WebGPU/TextureManager/index.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  vtkWebGPUTextureManager(publicAPI, model);
+}
+
+// ----------------------------------------------------------------------------
+
+export const newInstance = macro.newInstance(extend);
+
+// ----------------------------------------------------------------------------
+
+export default { newInstance, extend, ...STATIC };
```
Comment (jourdain) [inline Sources/Rendering/WebGPU/TextureManager/index.js:146]: no static

**Comment by jourdain** [comment]
Glancing through the code, it seems fine.

**Pull request merged**